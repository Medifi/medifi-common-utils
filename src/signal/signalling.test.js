import Signal from './index'

const user = {
  id: '1',
  fullName: 'Sieg Collado',
  role: 'Doctor'
}

const message = {
  id: '1',
  body: 'body'
}

describe('serializers', () => {
  const constantDate = new Date('2018-01-01T12:00:00')
  const RealDate = global.Date

  beforeAll(() => {
    global.Date = class extends Date {
      constructor () {
        super()
        return constantDate
      }
    }
  })

  describe('serializeSignal', () => {
    it('throws error on invalid user', () => {
      expect(() => {
        Signal.serializers.serializeSignal({ name: 'foo' }, new Date())
      }).toThrow('expected user to contain id, fullName and role')
    })
  })

  describe('message', () => {
    it('matches snapshot', () => {
      const result = Signal.serializers.message(user, new Date(), message)
      expect(result).toMatchSnapshot()
    })

    it('throws an error on invalid message', () => {
      expect(() => {
        Signal.serializers.message(user, new Date(), { id: '1', message: 'message!' })
      }).toThrow('expected message to contain id and body')
    })
  })

  describe('typing', () => {
    it('matches snapshot', () => {
      const result = Signal.serializers.typing(user, new Date(), true)
      expect(result).toMatchSnapshot()
    })

    it('matches snapshot on false', () => {
      const result = Signal.serializers.typing(user, new Date(), false)
      expect(result).toMatchSnapshot()
    })
  })

  describe('end-session', () => {
    it('matches snapshot', () => {
      const result = Signal.serializers.endSession(user, new Date(), true)
      expect(result).toMatchSnapshot()
    })
  })

  describe('rejecting a video invite', () => {
    it('matches snapshot', () => {
      const result = Signal.serializers.rejectVideoInvite(user, new Date())
      expect(result).toMatchSnapshot()
    })
  })

  describe('ending a video call', () => {
    it('matches snapshot', () => {
      const result = Signal.serializers.endVideoCall(user, new Date())
      expect(result).toMatchSnapshot()
    })
  })

  describe('connection', () => {
    it('matches snapshot', () => {
      const result = Signal.serializers.connection(user, new Date(), true)
      expect(result).toMatchSnapshot()
    })
  })
})

describe('parsing', () => {
  it('successfully parses a message signal', () => {
    const messageSignal = Signal.serializers.message(user, new Date(), message)
    const result = Signal.parseSignal(messageSignal)
    expect(result).toEqual(expect.objectContaining({ user, body: message.body, id: message.id }))
    expect(result).toMatchSnapshot()
  })

  it('parses a typing signal', () => {
    const typingSignal = Signal.serializers.typing(user, new Date(), true)
    const result = Signal.parseSignal(typingSignal)
    console.log('typing signal:', result)
    expect(result).toEqual(expect.objectContaining({ user, isTyping: true }))
    expect(result).toMatchSnapshot()
  })

  it('parses an end session signal', () => {
    const endSignal = Signal.serializers.endSession(user, new Date(), true)
    const result = Signal.parseSignal(endSignal)
    expect(result).toEqual(expect.objectContaining({ user }))
    expect(result).toMatchSnapshot()
  })

  it('parses a reject video invite signal', () => {
    const rejectSignal = Signal.serializers.rejectVideoInvite(user, new Date())
    const result = Signal.parseSignal(rejectSignal)
    expect(result).toEqual(expect.objectContaining({ user }))
    expect(result).toMatchSnapshot()
  })

  it('parses a end video call signal', () => {
    const endVideoCallSignal = Signal.serializers.endVideoCall(user, new Date())
    const result = Signal.parseSignal(endVideoCallSignal)
    expect(result).toEqual(expect.objectContaining({ user }))
    expect(result).toMatchSnapshot()
  })

  it('parses a connection signal', () => {
    const connectionSignal = Signal.serializers.connection(user, new Date(), true)
    const result = Signal.parseSignal(connectionSignal)
    expect(result).toEqual(expect.objectContaining({ user, ready: true }))
    expect(result).toMatchSnapshot()
  })
})

describe('getSerializer', () => {
  it('parses a signal message', () => {
    const serializer = Signal.serializers.getSerializer(Signal.types.MESSAGE)
    const messageSignal = serializer(user, new Date(), message)
    const result = Signal.parseSignal(messageSignal)
    expect(result).toEqual(expect.objectContaining({ user, body: message.body, id: message.id }))
  })
})
