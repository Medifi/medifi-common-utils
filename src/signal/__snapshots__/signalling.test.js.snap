// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parsing parses a connection signal 1`] = `
Object {
  "dateSent": 2018-01-01T04:00:00.000Z,
  "ready": true,
  "user": Object {
    "fullName": "Sieg Collado",
    "id": "1",
    "role": "Doctor",
  },
}
`;

exports[`parsing parses a end video call signal 1`] = `
Object {
  "dateSent": 2018-01-01T04:00:00.000Z,
  "user": Object {
    "fullName": "Sieg Collado",
    "id": "1",
    "role": "Doctor",
  },
}
`;

exports[`parsing parses a reject video invite signal 1`] = `
Object {
  "dateSent": 2018-01-01T04:00:00.000Z,
  "user": Object {
    "fullName": "Sieg Collado",
    "id": "1",
    "role": "Doctor",
  },
}
`;

exports[`parsing parses a request video call cancel signal 1`] = `
Object {
  "dateSent": 2018-01-01T04:00:00.000Z,
  "user": Object {
    "fullName": "Sieg Collado",
    "id": "1",
    "role": "Doctor",
  },
}
`;

exports[`parsing parses a request video call signal 1`] = `
Object {
  "dateSent": 2018-01-01T04:00:00.000Z,
  "user": Object {
    "fullName": "Sieg Collado",
    "id": "1",
    "role": "Doctor",
  },
}
`;

exports[`parsing parses a typing signal 1`] = `
Object {
  "dateSent": 2018-01-01T04:00:00.000Z,
  "isTyping": true,
  "user": Object {
    "fullName": "Sieg Collado",
    "id": "1",
    "role": "Doctor",
  },
}
`;

exports[`parsing parses an accept video invite signal 1`] = `
Object {
  "dateSent": 2018-01-01T04:00:00.000Z,
  "user": Object {
    "fullName": "Sieg Collado",
    "id": "1",
    "role": "Doctor",
  },
}
`;

exports[`parsing parses an end session signal 1`] = `
Object {
  "dateSent": 2018-01-01T04:00:00.000Z,
  "user": Object {
    "fullName": "Sieg Collado",
    "id": "1",
    "role": "Doctor",
  },
}
`;

exports[`parsing successfully parses a message signal 1`] = `
Object {
  "body": "body",
  "dateSent": 2018-01-01T04:00:00.000Z,
  "id": "1",
  "user": Object {
    "fullName": "Sieg Collado",
    "id": "1",
    "role": "Doctor",
  },
}
`;

exports[`serializers connection matches snapshot 1`] = `"{\\"ready\\":true,\\"dateSent\\":1514779200000,\\"user\\":{\\"id\\":\\"1\\",\\"fullName\\":\\"Sieg Collado\\",\\"role\\":\\"Doctor\\"}}"`;

exports[`serializers end-session matches snapshot 1`] = `"{\\"dateSent\\":1514779200000,\\"user\\":{\\"id\\":\\"1\\",\\"fullName\\":\\"Sieg Collado\\",\\"role\\":\\"Doctor\\"}}"`;

exports[`serializers message matches snapshot 1`] = `"{\\"id\\":\\"1\\",\\"body\\":\\"body\\",\\"dateSent\\":1514779200000,\\"user\\":{\\"id\\":\\"1\\",\\"fullName\\":\\"Sieg Collado\\",\\"role\\":\\"Doctor\\"}}"`;

exports[`serializers typing matches snapshot 1`] = `"{\\"isTyping\\":true,\\"dateSent\\":1514779200000,\\"user\\":{\\"id\\":\\"1\\",\\"fullName\\":\\"Sieg Collado\\",\\"role\\":\\"Doctor\\"}}"`;

exports[`serializers typing matches snapshot on false 1`] = `"{\\"isTyping\\":false,\\"dateSent\\":1514779200000,\\"user\\":{\\"id\\":\\"1\\",\\"fullName\\":\\"Sieg Collado\\",\\"role\\":\\"Doctor\\"}}"`;

exports[`serializers video call events accepting a video invite matches snapshot 1`] = `"{\\"dateSent\\":1514779200000,\\"user\\":{\\"id\\":\\"1\\",\\"fullName\\":\\"Sieg Collado\\",\\"role\\":\\"Doctor\\"}}"`;

exports[`serializers video call events cancelling a video call matches snapshot 1`] = `"{\\"dateSent\\":1514779200000,\\"user\\":{\\"id\\":\\"1\\",\\"fullName\\":\\"Sieg Collado\\",\\"role\\":\\"Doctor\\"}}"`;

exports[`serializers video call events ending a video call matches snapshot 1`] = `"{\\"dateSent\\":1514779200000,\\"user\\":{\\"id\\":\\"1\\",\\"fullName\\":\\"Sieg Collado\\",\\"role\\":\\"Doctor\\"}}"`;

exports[`serializers video call events initiating a video call matches snapshot 1`] = `"{\\"dateSent\\":1514779200000,\\"user\\":{\\"id\\":\\"1\\",\\"fullName\\":\\"Sieg Collado\\",\\"role\\":\\"Doctor\\"}}"`;

exports[`serializers video call events rejecting a video invite matches snapshot 1`] = `"{\\"dateSent\\":1514779200000,\\"user\\":{\\"id\\":\\"1\\",\\"fullName\\":\\"Sieg Collado\\",\\"role\\":\\"Doctor\\"}}"`;
